MetaMask/metamask-extension
# UX: Load the extension HTML pages and background with async JavaScript

## Explanation

At present the loading of both the fullscreen view and MetaMask popup are really slow due to the loading of ~15-20 synchronous JavaScript files in `home.html`, `popup.html`, and `notification.html`.  Our users experience this slowness most notably when clicking on the MetaMask icon in the extensions toolbar -- the click happens, then nothing for seemingly 1-2 seconds, and then the popup finally renders the user's wallet.

This PR loads both the UI and background scripts (1) sequentially and (2) asynchronously, drastically improving perceived performance.  Now when clicking the MetaMask extension icon, the popup displays *immediately*, shows a spinner while the scripts load, and then load in the UI.

## Screenshots/Screencaps

https://github.com/MetaMask/metamask-extension/assets/46655/23a83957-f4f4-45aa-a0db-9108867f84da

## Manual Testing Steps

1.  Pull down this PR
2. `yarn start`
3. Click the MetaMask logo in the extension toolbar
4. See the popup display *very quickly*, showing the MetaMask head and spinner, and roughly 1 second later the extension displays and functions as normal
5. Open the fullscreen view, see it load and work as normal

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
